numpy/lib/_datasource.py

```diff
@@ -38,9 +38,44 @@
 import os
 import sys
 import shutil
+import io
 
 _open = open
 
+def _bz2open(fn, mode, encoding, newline):
+    """ wrapper to open bz2 in text mode """
+    import bz2
+    if sys.version_info[0] >= 3:
+        return bz2.open(fn, mode=mode, encoding=encoding, newline=newline)
+
+    if "t" in mode:
+        raise UnsupportedMode("bz2 files not supported in python2")
+    else:
+        return bz2.BZ2File(fn, mode.replace("t", ""))
```

Comment:
yes
bz2 probably can't easily work in python2 as it is not compatible with TextIOWrapper. Luckily it is not very much used anymore.